api/oidc_test.go

```diff
@@ -0,0 +1,143 @@
+					http.StatusInternalServerError,
+				)
+				return
+			}
+
+			if !bytes.Equal(body, expectedBody) {
+				t.Errorf("wanted = %q, got = %q", expectedBody, body)
+				http.Error(
+					rw,
+					fmt.Sprintf(
+						`{"message:"Bad Request: wanted = %q, got = %q"}`,
+						expectedBody,
+						body,
+					),
+					http.StatusBadRequest,
+				)
+				return
+			}
+
+			io.WriteString(rw, fmt.Sprintf(`{"token":"%s"}`, oidcToken))
+
+		default:
+			http.Error(
+				rw,
+				fmt.Sprintf(
+					`{"message":"Not Found; method = %q, path = %q"}`,
+					req.Method,
+					req.URL.Path,
+				),
+				http.StatusNotFound,
+			)
+		}
+	}))
+}
+
+func TestOidcToken(t *testing.T) {
+	const jobId = "b078e2d2-86e9-4c12-bf3b-612a8058d0a4"
+	const oidcToken = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTUxNjIzOTAyMn0.NHVaYe26MbtOYhSKkoKYdFVomg4i8ZJd8_-RU8VNbftc4TSMb4bXP3l3YlNWACwyXPGffz5aXHc6lty1Y2t4SWRqGteragsVdZufDn5BlnJl9pdR_kdVFUsra2rWKEofkZeIC4yWytE58sMIihvo9H1ScmmVwBcQP6XETqYd0aSHp1gOa9RdUPDvoXQ5oqygTqVtxaDr6wUFKrKItgBMzWIdNZ6y7O9E0DhEPTbE9rfBo6KTFsHAZnMg4k68CDp2woYIaXbmYTWcvbzIuHO7_37GT79XdIwkm95QJ7hYC9RiwrV7mesbY4PAahERJawntho0my942XheVLmGwLMBkQ"
+	const accessToken = "llamas"
+
+	for _, testData := range []struct {
+		OidcTokenRequest *api.OidcTokenRequest
+		AccessToken      string
+		ExpectedBody     []byte
+		OidcToken        *api.OidcToken
+		Error            error
+	}{
+		{
+			AccessToken: accessToken,
+			OidcTokenRequest: &api.OidcTokenRequest{
+				JobId: jobId,
+			},
+			ExpectedBody: []byte("{}\n"),
+			OidcToken:    &api.OidcToken{Token: oidcToken},
+		},
+		{
+			AccessToken: accessToken,
+			OidcTokenRequest: &api.OidcTokenRequest{
+				JobId:    jobId,
+				Audience: "sts.amazonaws.com",
+			},
+			ExpectedBody: []byte(`{"audience":"sts.amazonaws.com"}
+`),
+			OidcToken: &api.OidcToken{Token: oidcToken},
+		},
+	} {
+		func() { // this exists to allow closing the server on each iteration
+			path := fmt.Sprintf("/jobs/%s/oidc/tokens", testData.OidcTokenRequest.JobId)
+
+			server := newOidcTokenServer(
+				t,
+				testData.AccessToken,
+				testData.OidcToken.Token,
+				path,
+				testData.ExpectedBody,
+			)
+			defer server.Close()
+
+			// Initial client with a registration token
+			client := api.NewClient(logger.Discard, api.Config{
+				UserAgent: "Test",
+				Endpoint:  server.URL,
+				Token:     accessToken,
+				DebugHTTP: true,
+			})
+
+			if token, resp, err := client.OidcToken(testData.OidcTokenRequest); err != nil {
+				if !errors.Is(err, testData.Error) {
+					t.Fatalf(
+						"OidcToken(%v) got error = %v, want error = %v",
+						testData.OidcTokenRequest,
+						err,
+						testData.Error,
+					)
+				}
+			} else if !cmp.Equal(token, testData.OidcToken) {
+				t.Fatalf("OidcToken(%v) got token = %v, want %v", testData.OidcTokenRequest, token, testData.OidcToken)
+			} else if resp.StatusCode != http.StatusOK {
```

Comment:
This `else` is redundant - the `t.Fatalf` would exit the test.

Consider checking both the response token and `resp.StatusCode`, with `t.Errorf`, but not in an `else` branch.